apps/explorer/src/pages/AddressInfoPage/NFTList.tsx

```diff
@@ -16,60 +16,135 @@ You should have received a copy of the GNU Lesser General Public License
+  }
+
   return (
-    <NFTListContainer>
-      {isLoading ? (
-        <NFTListStyled>
-          <SkeletonLoader height="200px" />
-          <SkeletonLoader height="200px" />
-          <SkeletonLoader height="200px" />
-        </NFTListStyled>
-      ) : nfts.length > 0 ? (
-        <NFTListStyled>
-          {nfts.map((nft) => (
-            <NFTItem key={nft.id} nft={nft} />
-          ))}
-        </NFTListStyled>
-      ) : (
-        <NoNFTsMessage>
-          <EmptyIconContainer>
-            <RiGhostLine />
-          </EmptyIconContainer>
-          <div>{t('No NFTs yet')}</div>
-        </NoNFTsMessage>
-      )}
-    </NFTListContainer>
+    <>
+      <NFTListContainer>
+        <Toolbar>
+          <ToggleLabel>{t('Group by collection')}</ToggleLabel>
+          <Toggle
+            label={t('Group by collection')}
+            toggled={isCollectionGroupingActive === 'on'}
+            onToggle={handleCollectionToggle}
+          />
+        </Toolbar>
+        {isLoading ? (
+          <NFTListStyled>
+            <SkeletonLoader height="200px" />
+            <SkeletonLoader height="200px" />
+            <SkeletonLoader height="200px" />
+          </NFTListStyled>
+        ) : nfts.length > 0 ? (
+          isCollectionGroupingActive === 'on' ? (
+            Object.entries(NFTsGroupedByCollection).map(([collectionId, nfts]) => (
+              <CollectionContainer key={collectionId}>
+                <CollectionHeader>
+                  {collectionId === 'undefined'
+                    ? t('Unknown collection')
+                    : collectionFiles.find((c) => c.collectionId === collectionId)?.name}
+                </CollectionHeader>
+                <NFTListComponent nfts={nfts} onClick={setConsultedNftId} />
+              </CollectionContainer>
+            ))
+          ) : (
+            <NFTListComponent nfts={nfts} onClick={setConsultedNftId} />
+          )
+        ) : (
+          <NoNFTsMessage>
+            <EmptyIconContainer>
+              <RiGhostLine />
+            </EmptyIconContainer>
+            <div>{t('No NFTs yet')}</div>
+          </NoNFTsMessage>
+        )}
+      </NFTListContainer>
+      <ModalPortal>
+        <NFTDetailsModal
+          nft={consultedNft}
+          collection={collectionFiles.find((c) => c.collectionId === consultedNft?.collectionId)}
+          isOpen={!!consultedNftId}
+          onClose={() => setConsultedNftId(undefined)}
+        />
+      </ModalPortal>
+    </>
   )
 }
 
+interface NFTListComponentProps {
+  nfts?: NFTMetadataWithFile[]
+  onClick: (nftId: string) => void
+}
+
+const NFTListComponent = ({ nfts, onClick }: NFTListComponentProps) => (
+  <NFTListStyled>
+    {nfts ? nfts.map((nft) => <NFTItem key={nft.id} nft={nft} onClick={() => onClick(nft.id)} />) : null}
+  </NFTListStyled>
+)
+
 interface NFTItemProps {
-  nft: UnverifiedNFTMetadataWithFile
+  nft: NFTMetadataWithFile
+  onClick: (nftId: string) => void
```

Comment:
Since this component can only render a single NFT, the `nftId` of this callback will always be the ID of what we pass in the `nft` prop. It would make sense to parametrize the callback when we pass a list of NFTs. But for a single one, it's not needed. 

You anyways always pass a callback bound to that specific NFT id when you instantiate the component (like 136)

You can remove the `nftId` prop here and you can simplify the code below (see next comment)

Reply:
Tried multiple approaches and didn't clean up. Thanks.